Harlem date bug fix

jonharlem opened the pull request:
The current bug is in the master codebase already -- NoMethodError in Quotes#index. This behavior results because 2 of the quotes in the current migration do not have created_at or updated_at timestamps. I fixed the bug by writing an if/else statement to account for use cases where the quotes don't have timestamps.

jonharlem committed:
added if/else statement to account for no created_at date

jonharlem committed:
test for create_at bug

jonharlem committed:
added test for rendering div.created

jonharlem committed:
added test for relative date 2 days ago

harms280 commented:
Hey Jon,

  Thanks for working on this. You should add some detail on what was throwing the bug initially (what a user had to do in order to come across the error). Otherwise it looks like you successfully pinpointed what was throwing the error. You are on the right track to fixing the bug, but instead of using a conditional on the view, we would prefer if none of the data in the database had nil as a value for it's creation and updated properties. Could you change the production.sql data so that the values of the dates are not nil? Also, can you make it so the quote model will validate the presence of a created_at and updated_at field so that it will never have the value of nil? 

Thanks!

jonharlem committed:
added validation and modified test appropriately

jonharlem commented:
The user would log in with a correct username and password, but would not be able to see the rendered site and would instead run into NoMethodError in Quotes#index. According to feedback, I added validation so that the database would never have a value of nil and I changed the production.sql data so that there were no nils.

harms280 commented:
Very good, everything is looking great and has been merged.
